File: compute_science/selectionSort.js
function swap(largerIndex, smallerIndex, arr) {
  const temp = arr[smallerIndex];
  arr[smallerIndex] = arr[largerIndex];
  arr[largerIndex] = temp;
}
let testArr = [1, 2, 3, 12, 0, -2, 4, 2];

function selectionSort(arr) {
  for (let i = 0; i < arr.length; i++) {
    let minIndex = i;
    for (let j = i; j < arr.length; j++) {
      if (arr[j] < arr[minIndex]) {
        minIndex = j;
      }
    }
    if (i !== minIndex) swap(i, minIndex, arr);
  }
  return arr;
}

console.log(selectionSort(testArr));

File: compute_science/selection_sort.py
def swap(left, right, arr):
    temp = arr[left]
    arr[left] = arr[right]
    arr[right] = temp


def selection_sort(arr: list[int]) -> list[int]:
    for i in range(len(arr)):
        minIndex = i
        for j in range(i, len(arr)):
            if arr[j] < arr[minIndex]:
                minIndex = j
        swap(minIndex, i, arr)
    return arr


testArr = [4, 6, 1, 2, 90, 55, -1, -11]
selection_sort(testArr)
print(testArr)
